thinc/layers/softmax.py

```diff
@@ -17,29 +17,36 @@ def Softmax(
     nI: Optional[int] = None,
     *,
     init_W: Callable = zero_init,
-    init_b: Callable = zero_init
+    init_b: Callable = zero_init,
+    normalize: bool = True,
 ) -> Model[InT, OutT]:
     return Model(
         "softmax",
         forward,
         init=partial(init, init_W, init_b),
         dims={"nO": nO, "nI": nI},
         params={"W": None, "b": None},
+        attrs={"normalize": normalize},
     )
 
 
 def forward(model: Model[InT, OutT], X: InT, is_train: bool) -> Tuple[OutT, Callable]:
+    normalized = model.attrs["normalize"] or is_train
     W = cast(Floats2d, model.get_param("W"))
     b = cast(Floats1d, model.get_param("b"))
     Y = model.ops.affine(X, W, b)
-    Y = model.ops.softmax(Y)
+    if normalized:
+        Y = model.ops.softmax(Y)
 
     def backprop(dY: InT) -> OutT:
         model.inc_grad("b", dY.sum(axis=0))
         model.inc_grad("W", model.ops.gemm(dY, X, trans1=True))
         return model.ops.gemm(dY, W)
 
-    return Y, backprop
+    if normalized:
+        return Y, backprop
+    else:
+        return Y, lambda dY: []
```

Comment:
```suggestion
        return Y, lambda dY: []
```

Better to return `ops.xp.zeroslike(dY)` or raise an error I think. The empty list will give type errors or confusing situations.

Reply:
Raised an error, better to notice when someone tries to backprop into an unnormalized softmax.

We could also consider computing the normalized probabilities when someone tries to backprop. It would give the best of both worlds: fast, unnormalized softmax during inference and training still works.